### DISCUSSION TITLE
End training block when a criterion is met

### DISCUSSION THREAD

**Discussion by lapensee** [jspsych/jsPsych#1291] [Q&A] [answered]
Hi,

I am trying to code an experiment where one learns a category while getting feedback, and they also learn up to a criterion (e.g., more than 8 trials correct in the last 10 trials) before getting tested.

To manage the feedback part, I've set up a loop with a learning trial and a feedback trial, which has a jsPsych.get function to evaluate the response made in the learning trial.  

However, I wasn't sure how to implement the part where the learning stops when a certain criterion is met.

I've tried to have a running average vector and save that in data in the learning trial (by using a queue-like mechanisms).  Then in the feedback trial there is an on_finish function that evaluates the running average vector which turns the for loop index to something that will make the loop stop.   I think I am not used to the fact that javascript can't do things on the fly and this method isn't conceptually working.....

Would there be any other ways/examples that I can use to implement this experiment?

I would appreciate any advice!  Thank you!

Best,
Hyungwook

**Comment by jodeleeuw** [comment, marked as answer]
Hi @lapensee 

Check out this thread: https://github.com/jspsych/jsPsych/discussions/1245

**Reply by lapensee** [reply]
Thank you so much!
